lib/promiscuous/redis.rb

```
@@ -158,6 +158,11 @@ def try_lock
       !!result
     end
 
+    def extend
+      @expires_at = Time.now + @expire + 1
+      @node.set(@key, "#{@expires_at}:#{@token}")
```

This is racy with someone recovering the lock.

You can only set it if `"#{@expires_at}:#{@token}"` has the same old value. If not, you can raise a LostLock exception or something.